### ISSUE TITLE
inspect.ismethoddescriptor(): lack of __delete__() is not checked

### ISSUE THREAD

**Issue by zuo** [python/cpython#120381]
# Bug report

### Bug description:

The `inspect.ismethoddescriptor()` function reports descriptor objects which implement `__delete__()` but not `__set__()` as *method descriptors*, even though they are, in fact, *data descriptors*:

Actual behavior example:

```python
>>> import inspect
>>> class Descriptor:
...     def __get__(self, *_): pass
...     def __delete__(self, *_): pass  # Note: `__set__()` *not* defined.
...
>>> inspect.ismethoddescriptor(Descriptor())  # Wrong result:
True
```

Expected result:

```python
>>> inspect.ismethoddescriptor(Descriptor())  # Correct result:
False
```

See also: https://discuss.python.org/t/inspect-ismethoddescriptor-checks-for-the-lack-of-set-but-ignores-delete-should-it-be-fixed/55039

***

There is an additional question: *to which Python versions should the fix be applied?*

IMHO, the fix can be safely applied to the branches 3.14 (main) and 3.13 (still in the beta phase). Backporting it to the earlier versions might be considered too disruptive. Obviously, the decision will need to be made by a core developer, not me.

***

### CPython versions tested on:

3.8, 3.9, 3.10, 3.11, 3.12, 3.13, CPython main branch

### Linked PRs
* gh-120383
* gh-120684

**Comment by ncoghlan** [comment]
Merged and fixed for the next 3.13 beta. Thanks for the report and PR, @zuo!

**Comment by zuo** [comment]
> Merged and fixed for the next 3.13 beta. Thanks for the report and PR, @zuo!

Thank you too! :)
